controller/controller.go

```diff
@@ -29,10 +29,10 @@ type Controller struct {
 	Consensus *bft.BFT          // the async consensus process between the committee members for the chain
 	P2P       *p2p.P2P          // the P2P module the node uses to connect to the network
 
-	RootChainInfo lib.RootChainInfo // the latest information from the 'root chain'
-	isSyncing     *atomic.Bool      // is the chain currently being downloaded from peers
-	log           lib.LoggerI       // object for logging
-	sync.Mutex                      // mutex for thread safety
+	RCManager  lib.RCManagerI // the rpc client for the 'root chain'
```

Comment:
nitpick: I see the word `rpc` used multiple times to refer to the `RCManager` which under the hood uses a Websockets client which is based on `http` not `rpc`, is just a bit misleading unless you may refer to something else